buildscripts/Set-VersionInfo.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+
+.SYNOPSIS
+
+Extracts build version information from Settings.proj and the
+environment. Then constructs several flavor version strings and
+uses them, along with the current year to
+
+  - update buildscripts/CommonAssemblyInfo.cs,
+  - update all project.json files, and
+  - emit TeamCity directives so it can know about the values
+
+#>
+
+[xml] $settings = Get-Content 'Settings.proj'
+$version_group = $settings.Project.PropertyGroup[1]
+
+$major = $version_group.Project_Major
+$minor = $version_group.Project_Minor
+$build = $version_group.Project_Build
```

Comment:
I know the old script named this "build", but I think should should align with semver and refer to it as "patch" so it isn't confusing with "build number"